nixos/modules/system/boot/stage-1-init.sh

```diff
@@ -334,8 +334,10 @@ mountFS() {
 
     # Filter out x- options, which busybox doesn't do yet.
     local optionsFiltered="$(IFS=,; for i in $options; do if [ "${i:0:2}" != "x-" ]; then echo -n $i,; fi; done)"
+    # Prefix (lower|upper|work)dir with /mnt-root (overlayfs)
+    local optionsPrefixed="$(IFS=,; for i in $optionsFiltered; do case ${i%=*} in (lowerdir|upperdir|workdir) echo -n "${i%=*}=/mnt-root${i##*=}," ;; (*) echo -n "${i}," ;; esac; done)"
```

Comment:
Here's some sed magic to do the same with a bunch of chars and head-scratching saved:
```suggestion
    local optionsPrefixed="$(sed -E 's#\<(lowerdir|upperdir|workdir)=#\1=/mnt-root#g' <<< "$optionsFiltered")"
```

Reply:
Updated with slightly modified version as `<<<` doesn't seem to work in ash.